app/parse.py

```diff
@@ -15,9 +26,266 @@ class Product:
+                    ".row.ecomerce-items.ecomerce-items-more"
+                ))
+            )
+            print("CLICK MORE BTN ", count_more_btn)
+            count_more_btn += 1
+        except ElementClickInterceptedException:
+            close_cookie_banner(driver)
+        except ElementNotInteractableException:
+            print("exit from loop")
+            break
+
+    print("we are waiting")
+    WebDriverWait(driver, 10).until(
+        expected_conditions.presence_of_element_located((
+            By.CSS_SELECTOR, ".col-md-4.col-xl-4.col-lg-4"
+        ))
+    )
+    print("we are taking all cards")
+
+    cards = driver.find_elements(
+        By.CSS_SELECTOR, ".col-md-4.col-xl-4.col-lg-4"
+    )
+
+    print("we will start a loop")
+
+    for card in cards:
+        products.append(
+            Product(
+                title=card.find_element(
+                    By.CLASS_NAME,
+                    "title"
+                ).get_attribute("title"),
+                description=card.find_element(
+                    By.CLASS_NAME,
+                    "description"
+                ).text,
+                price=float(
+                    card.find_element(
+                        By.CLASS_NAME,
+                        "price"
+                    ).text.replace("$", "")
+                ),
+                rating=len(
+                    card.find_elements(
+                        By.CSS_SELECTOR,
+                        ".ratings p span.ws-icon"
+                    )
+                ),
+                num_of_reviews=int(
+                    card.find_element(
+                        By.CSS_SELECTOR,
+                        ".ratings > .review-count.float-end.pull-right"
+                    ).text.split()[0]
+                ),
+            )
+        )
+
+    return products
+
+
 def get_all_products() -> None:
-    pass
+    options = webdriver.ChromeOptions()
+    options.add_argument("--headless")
+    with webdriver.Chrome(options=options) as driver:
+        home_page_products = get_products_from_home_page(HOME_URL, driver)
+        write_products_to_csv_file(home_page_products, "home.csv")
+
+        computer_page = get_random_products(
+            HOME_URL, driver, "a[href='/test-sites/e-commerce/more/computers']"
+        )
+        write_products_to_csv_file(computer_page, "computers.csv")
+
+        laptop_page = get_all_products_from_page(
+            HOME_URL,
+            driver,
+            "a[href='/test-sites/e-commerce/more/computers']",
+            "a[href='/test-sites/e-commerce/more/computers/laptops']",
+        )
+        write_products_to_csv_file(laptop_page, "laptops.csv")
+
+        tablets_page = get_all_products_from_page(
+            HOME_URL,
+            driver,
+            "a[href='/test-sites/e-commerce/more/computers']",
+            "a[href='/test-sites/e-commerce/more/computers/tablets']",
+        )
+        write_products_to_csv_file(tablets_page, "tablets.csv")
```

Comment:
The function `write_products_to_csv_file` is called here, but it is not defined within the provided code. This will result in a `NameError`. Ensure that this function is defined or imported correctly.